src/main/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/resource/UserGroupController.kt

```diff
@@ -0,0 +1,94 @@
+package uk.gov.justice.digital.hmpps.externalusersapi.resource
+
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.Parameter
+import io.swagger.v3.oas.annotations.media.Content
+import io.swagger.v3.oas.annotations.media.Schema
+import io.swagger.v3.oas.annotations.responses.ApiResponse
+import io.swagger.v3.oas.annotations.responses.ApiResponses
+import io.swagger.v3.oas.annotations.tags.Tag
+import org.slf4j.LoggerFactory
+import org.springframework.http.HttpStatus
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.DeleteMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.ResponseStatus
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.externalusersapi.config.ErrorResponse
+import uk.gov.justice.digital.hmpps.externalusersapi.service.UserGroupService
+
+@RestController
+@Tag(name = "/user/id/{userId}/groups", description = "User Groups Controller")
+class UserGroupController(
+  private val userGroupService: UserGroupService
+) {
+
+  companion object {
+    private val log = LoggerFactory.getLogger(this::class.java)
+  }
+
+  @DeleteMapping("/user/id/{userId}/groups/{group}")
+  @ResponseStatus(HttpStatus.NO_CONTENT)
+  @PreAuthorize("hasAnyRole('ROLE_MAINTAIN_OAUTH_USERS', 'ROLE_AUTH_GROUP_MANAGER')")
+  @Operation(
+    summary = "Remove group from user.",
+    description = "Remove group from user."
+  )
+  @ApiResponses(
+    value = [
+      ApiResponse(
+        responseCode = "204",
+        description = "Deleted"
+      ),
+      ApiResponse(
+        responseCode = "400",
+        description = "Validation failed.",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class)
```

Comment:
Not sure if removing a group is "validation failed"? Is this validation failed on the param passed in?

Reply:
As discussed earlier I will raise a ticket to look at this separately, in addition to this end point giving a 200 response when the user id is not recognised, plus converting it to a UUID in the controller.